realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -237,6 +240,16 @@ public RxObservableFactory getRxFactory() {
         return rxObservableFactory;
     }
 
+    /**
+     * Returns whether this Realm is read only or not. Read-only Realms cannot be modified and will throw an
+     * {@link IllegalStateException} if {@link Realm#beginTransaction()} is called on it.
+     *
+     * @return {@code true} if this Realm is read only, {@code false} if not.
+     */
+    public boolean isReadOnly() {
+        return readOnly;
+    }
+
     @Override
     public boolean equals(Object obj) {
```

Comment:
So it is allowed that user to create a read-only realm in thread A, but writeable Realm in thread B? I don't think this matches the behaviour from Object Store -- we need to ensure the behaviour is the same after we migrate to the Object Store schema, otherwise there is going to be a breaking change.